From 0cae93e87e136fcc9ff1ecc9563ff2075bff0346 Mon Sep 17 00:00:00 2001
From: Nariman Poushin <nariman@opensource.wolfsonmicro.com>
Date: Thu, 13 Feb 2014 01:56:53 +0000
Subject: [PATCH 127/351] switch: arizona: Move the hpdet_cb invocations to be
 more robust

Move the invocation of the callback as close to the impedance
measurement / jack removal as possible to make sure we don't
miss any events.

Change-Id: If003ef3417a0b6c8c6e76d501a46a68a19609395
Signed-off-by: Nariman Poushin <nariman@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 drivers/switch/switch-arizona.c |    7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/switch/switch-arizona.c b/drivers/switch/switch-arizona.c
index 9869eb5..6160790 100644
--- a/drivers/switch/switch-arizona.c
+++ b/drivers/switch/switch-arizona.c
@@ -500,6 +500,10 @@ static int arizona_hpdet_read(struct arizona_extcon_info *info)
 	}
 
 	info->hp_impedance = val;
+
+	if (arizona->pdata.hpdet_cb)
+		arizona->pdata.hpdet_cb(info->hp_impedance);
+
 	dev_dbg(arizona->dev, "HP impedance %d ohms\n", val);
 	return val;
 }
@@ -768,9 +772,6 @@ static irqreturn_t arizona_hpdet_irq(int irq, void *data)
 		break;
 	}
 
-	if (arizona->pdata.hpdet_cb)
-		arizona->pdata.hpdet_cb(reading);
-
 	if (mic || info->mic)
 		switch_set_state(&info->edev, BIT_HEADSET);
 	else
-- 
1.7.9.5

